src/bin/cargo/commands/uninstall.rs

```diff
@@ -37,3 +44,34 @@ pub fn exec(gctx: &mut GlobalContext, args: &ArgMatches) -> CliResult {
     ops::uninstall(root, specs, &values(args, "bin"), gctx)?;
     Ok(())
 }
+
+fn get_installed_crates() -> Vec<clap_complete::dynamic::CompletionCandidate> {
+    let gctx = match GlobalContext::default() {
+        Ok(gctx) => gctx,
+        Err(_) => return vec![],
+    };
+    let mut candidates = Vec::new();
```

Comment:
What I might do in a situation like this is

```suggestion
fn get_installed_crates() -> Vec<clap_complete::dynamic::CompletionCandidate> {
    get_installed_crates_().unwrap_or_default()
}

fn get_installed_crates_() -> Option<Vec<clap_complete::dynamic::CompletionCandidate>> {
    let gctx = GlobalContext::default().ok()?;
```

That will allow you to flatten the whole function because you don't have to explicitly handle the error cases